pkg/controller/v1beta1/inferenceservice/rawkube_controller_test.go

```diff
@@ -2130,4 +2243,407 @@ var _ = Describe("v1beta1 inference service controller", func() {
+									},
+									Resources: v1.ResourceRequirements{
+										Limits: v1.ResourceList{
+											v1.ResourceCPU:    resource.MustParse(constants.OauthProxyResourceCPULimit),
+											v1.ResourceMemory: resource.MustParse(constants.OauthProxyResourceMemoryLimit),
+										},
+										Requests: v1.ResourceList{
+											v1.ResourceCPU:    resource.MustParse(constants.OauthProxyResourceCPURequest),
+											v1.ResourceMemory: resource.MustParse(constants.OauthProxyResourceMemoryRequest),
+										},
+									},
+									VolumeMounts: []v1.VolumeMount{
+										{
+											Name:      "proxy-tls",
+											MountPath: "/etc/tls/private",
+										},
+									},
+									TerminationMessagePath:   "/dev/termination-log",
+									TerminationMessagePolicy: "File",
+									ImagePullPolicy:          "IfNotPresent",
+								},
+							},
+							Volumes: []v1.Volume{
+								{
+									Name: "proxy-tls",
+									VolumeSource: v1.VolumeSource{
+										Secret: &v1.SecretVolumeSource{
+											SecretName:  serviceName,
+											DefaultMode: func(i int32) *int32 { return &i }(420),
+										},
+									},
+								},
+							},
+							ServiceAccountName:            constants.KserveServiceAccountName,
+							DeprecatedServiceAccount:      constants.KserveServiceAccountName,
+							SchedulerName:                 "default-scheduler",
+							RestartPolicy:                 "Always",
+							TerminationGracePeriodSeconds: &gracePeriod,
+							DNSPolicy:                     "ClusterFirst",
+							SecurityContext: &v1.PodSecurityContext{
+								SELinuxOptions:      nil,
+								WindowsOptions:      nil,
+								RunAsUser:           nil,
+								RunAsGroup:          nil,
+								RunAsNonRoot:        nil,
+								SupplementalGroups:  nil,
+								FSGroup:             nil,
+								Sysctls:             nil,
+								FSGroupChangePolicy: nil,
+								SeccompProfile:      nil,
+							},
+						},
+					},
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "RollingUpdate",
+						RollingUpdate: &appsv1.RollingUpdateDeployment{
+							MaxUnavailable: &intstr.IntOrString{Type: 1, IntVal: 0, StrVal: "25%"},
+							MaxSurge:       &intstr.IntOrString{Type: 1, IntVal: 0, StrVal: "25%"},
+						},
+					},
+					RevisionHistoryLimit:    &revisionHistory,
+					ProgressDeadlineSeconds: &progressDeadlineSeconds,
+				},
+			}
+			Expect(actualDeployment.Spec).To(gomega.Equal(expectedDeployment.Spec))
+
+			//check service
+			actualService := &v1.Service{}
+			predictorServiceKey := types.NamespacedName{Name: constants.PredictorServiceName(serviceKey.Name),
+				Namespace: serviceKey.Namespace}
+			Eventually(func() error { return k8sClient.Get(context.TODO(), predictorServiceKey, actualService) }, timeout).
+				Should(Succeed())
+
+			expectedService := &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      predictorServiceKey.Name,
+					Namespace: predictorServiceKey.Namespace,
```

Comment:
Add unit tests for the routes.

Reply:
The route logic is checking if the route is `admitted` and we will only use the route URL if that is the case. Since ODH Model Controller is the one creating the route here shouldn't this be a integ test more than a unit test?

Reply:
Then, perhaps you should write an E2E test under [test/e2e](https://github.com/opendatahub-io/kserve/tree/master/test/e2e).

The E2Es are currently failing because of a bad openshift-ci setup. But once the setup is finished (we do have Jiras in the backlog for fixing it), the E2Es will run with odh-model-controller deployed on the testing cluster. In the meanwhile, we have been running E2Es on our machines.